### PR TITLE
Refactor CI to have similar flow for testing unsigned and signed binaries

### PR DESCRIPTION
**Pull request by SergeyKleyman**

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -112,10 +112,6 @@ pipeline {
                 }
               }
               stage('Static analysis and unit tests') {
-                when {
-                  beforeAgent true
-                  not { expression { isInternalCI() } }
-                }
```
Comment (v1v) [inline .ci/Jenkinsfile:118]: What's the reason to enable back this to the release?

File: .ci/Jenkinsfile [hunk 2]
```diff
@@ -478,110 +481,115 @@ def prepareRelease() {
...
 
-def generateSteps(file) {
-  def parallelTasks = [:]
-  dir("${BASE_DIR}"){
-    def output = sh(label: ".ci/${file}", script: ".ci/${file}", returnStdout: true)
-    if (output?.trim()) {
-      output.split('\n').each { content ->
-        parallelTasks[content] = generateStep(content)
+def generatePackageLifecycleTestMatrix(def args = [:]) {
+    def parallelTasks = [:]
```
Comment (v1v) [inline .ci/Jenkinsfile:485]: NIT:
ident of 2 rather than 4 for this function🙏

File: .ci/generate_package_lifecycle_test_matrix.sh [hunk 3]
```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+set -e
+
+#
+# Lifecycle tests
+#
+testingType=lifecycle
+for phpVersion in 7.2 7.3 7.4 8.0 8.1
```
Comment (v1v) [inline .ci/generate_package_lifecycle_test_matrix.sh:8]: What do you think to add a note here regarding this list of versions should be also reflected in the `.ci/Jenkinsfile` and viceversa?

I see references in https://github.com/elastic/apm-agent-php/blob/main/.ci/Jenkinsfile#L94 and https://github.com/elastic/apm-agent-php/blob/c4de75775741faf2e6ae6fa43574320a722b8336/.ci/Jenkinsfile#L165

**Review by v1v — APPROVED** [review]
LGTM, a couple of minor NITs.

OTOH, do you think we could add some references about how to add a new PHP version or stage in the LifecycleTestMatrix if needed? I'm more than happy to work on this with you .

**Comment by SergeyKleyman** [comment]
> OTOH, do you think we could add some references about how to add a new PHP version or stage in the LifecycleTestMatrix if needed? I'm more than happy to work on this with you .

It's a good idea - I've created an issue to follow up on this part - https://github.com/elastic/apm-agent-php/issues/770
I think just adding a new PHP version to `.ci/generate_package_lifecycle_test_matrix.sh` should be enough to have it tested by CI (of course assuming that support for the new version is already implemented).

File: .ci/Jenkinsfile (hunk 1 above)
Comment (SergeyKleyman) [reply .ci/Jenkinsfile:118]: I prefer to test the release just as thoroughly as PRs to have high confidence that nothing slipped through the cracks.

File: .ci/Jenkinsfile (hunk 2 above)
Comment (SergeyKleyman) [reply .ci/Jenkinsfile]: Done - https://github.com/elastic/apm-agent-php/pull/769/commits/c7f67bcb9c588a649010a6f61c778d1ba419f75e

**Pull request merged**